eliot/_parse.py

```diff
@@ -92,6 +114,63 @@ def add(self, message_dict):
             # Special case where there is no action:
             if written_message.task_level.level == [1]:
                 return self.transform(
-                    ["_nodes", self._root_level], written_message)
+                    ["_nodes", self._root_level], written_message,
+                    ["_completed"], lambda s: s.add(self._root_level))
             else:
                 return self._ensure_node_parents(written_message)
+
+
+class Parser(PClass):
+    """
+    Parse serialized Eliot messages into L{Task} instances.
+
+    @ivar _tasks: Map from UUID to corresponding L{Task}.
+    """
+    _tasks = pmap_field(unicode, Task)
+
+    def add(self, message_dict):
+        """
+        Update the L{} with a dictionary containing a serialized Eliot
+        message.
+
+        @param message_dict: Dictionary of serialized Eliot message.
+
+        @return: Tuple of (list of completed L{Task} instances, updated L{Parser}).
+        """
+        uuid = message_dict[TASK_UUID_FIELD]
+        if uuid in self._tasks:
```

Comment:
:grey_question: optional: `task = self._tasks.get(uuid, Task())` instead of this conditional.